strategy/compose_strategy.ts

```diff
@@ -1,25 +1,16 @@
 // Copyright 2018-2022 Gamebridge.ai authors. All rights reserved. MIT license.
-import { JSONValue } from "../serializable.ts";
 
 /** Functions used when hydrating data */
 // deno-lint-ignore no-explicit-any
-export type FromJSONStrategy = (value: JSONValue) => any;
-
-/** Functions used when dehydrating data */
-// deno-lint-ignore no-explicit-any
-export type ToJSONStrategy = (value: any) => JSONValue;
+export type Strategy = (value: any) => any;
```

Comment:
What's the reasoning behind this change? With how the previous strategies implementation worked we were able to enforce that input and output be formatted a certain way that we know will be consistent and usable JSON, with this new way the types are less restricted.

For example, now a user can create a toJson strategy where the output is something not serializable, like `Symbol()` or a `function`, resulting in what was previously a TS compile error now being presented as valid.